module/userProjects/validator/v1.js

```diff
@@ -6,38 +6,36 @@
  */
 
 module.exports = (req) => {
+	let projectsValidator = {
+		sync: function () {
+			req.checkParams('_id').exists().withMessage('required project id')
+			req.checkQuery('lastDownloadedAt').exists().withMessage('required last downloaded at')
+		},
+		tasksStatus: function () {
+			req.checkParams('_id').exists().withMessage('required project id')
+		},
+		solutionDetails: function () {
+			req.checkParams('_id').exists().withMessage('required project id')
+			req.checkQuery('taskId').exists().withMessage('required task id')
+		},
+		add: function () {
+			req.checkBody('title').exists().withMessage('required project title')
+			req.checkBody('categories[0].value').exists().withMessage('required categories[0].value')
```

Comment:
@MallanagoudaB This is not the way to do, what if categories[1].value is empty? Also we should check ids in DB, they cannot give any random ids

Reply:
Okay i have added a validator if the id is not present in DB will throw {
    "status": 400,
    "message": "Not found category"
}